opal/corelib/kernel.rb

```diff
@@ -779,16 +779,33 @@ def instance_variable_get(name)
   def instance_variable_set(name, value)
     name = Opal.instance_variable_name!(name)
 
-    `self[name.substr(1)] = value`
+    `self[Opal.ivar(name.substr(1))] = value`
   end
 
+  %x{
+    function restoreIvar(name) {
+      if (name === "constructor$" ||
+        name === "__proto__$" ||
+        name === "__parent__$" ||
+        name === "__noSuchMethod__$" ||
+        name === "__count__$" ||
+        name === "hasOwnProperty$" ||
+        name === "valueOf$")
+      {
+        return name.slice(0, name.length - 1);
+      }
+
+      return name;
+    }
+  }
+
```

Comment:
Initially I thought that we should not just blindly return all own properties that match some pattern. I mean
``` ruby
obj = Object.new
`#{obj}.attribute$ = 'some value'`
obj.instance_variables
```
should return a blank array, right? There are only a few reserved words in JS that must be explicitly escaped. I can revert this changes if you think that we can ignore such cases.